### PR TITLE
Design Picker: Add free tier filter to Type category in design picker

### PR DESCRIPTION
**Pull request by madhusudhand**
Closes: https://github.com/Automattic/dotcom-forge/issues/10091

## Proposed Changes

* Add `Free` option under Type category grouping
* Remove `Free only` radio component
* Remove the query param `tier` and make the tier value part of `categories` query param

image

image

## Testing Instructions

* Visit design picker: `setup/site-setup/designSetup?siteSlug=:site&siteId=:id`

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/design-picker/src/constants.ts [hunk 1]
```diff
@@ -45,6 +46,13 @@ export const FEATURE_CATEGORIES = {
 	STORE: 'store',
 };
 
+export const DESIGN_TIER_CATEGORIES = {
+	FREE: {
+		slug: FREE_THEME,
+		name: translate( 'Free' ),
```
Comment (arthur791004) [inline packages/design-picker/src/constants.ts]: The translation may not work before the locale is initialized.

**Review by fushar — COMMENTED** [review]
Selecting only the `Free` "filter" results in empty designs; I think it's unexpected:

image

**Comment by arthur791004** [comment]
**UPDATE**

Since there are some issues as follows after we move the `free-only` toggle into types, this PR focuses on the UI change and **all functionalities remain the same**.
* Do we need to show the `Free themes` section because it's a type?
* If we have a `Free themes` section
  * How can we remove the duplication of themes shown on multiple sections? Currently, a theme will be put into last matched selected category, so the result may look weird if the `Free` type is not the last selected one.
  * It may look weird that we still display non-free themes.
* If not, then the `Free` type works very differently

**Review by taipeicoder — APPROVED** [review]
Works as intended 👍

**Comment by taipeicoder** [comment]
> **UPDATE**
> 
> Since there are some issues as follows after we move the `free-only` toggle into types, this PR focuses on the UI change and **all functionalities remain the same**.
> 
> * Do we need to show the `Free themes` section because it's a type?
> * If not, then the `Free` type works very differently (current behavior)
> * If we have a `Free themes` section
>   
>   * How can we remove the duplication of themes shown on multiple sections? Currently, a theme will be put into last matched selected category, so the result may look weird if the `Free` type is not the last selected one.
>   * It may look weird that we still display non-free themes in other sections (e.g.: Store themes).

@fditrapani in case you missed this ☝️

**Comment by fditrapani** [comment]
Thanks for the ping @taipeicoder. I thought since it merged we'd be fine. To keep things simple, I think we should treat Free like the other filters. Do you all see any issues with that?

**Comment by taipeicoder** [comment]
An issue of having the Free filter behaving like others is that if users select the Free filter, then there will be a grouping section called Free where the themes in it won't be separated by category. There is also the issue of theme deduping, where if Theme A already shows up in the Recommended list, then it will not show in the Free group section.

A lot of conflicting logic makes this more complex than necessary. The Free filter really only works well as a toggle, similar to what's implemented in Webflow https://webflow.com/templates/all. That said, stakeholders were explicitly again introducing too many control components to the UI.

IMO, it's tricky to get it working as a Type filter. Maybe as a Collection filter as we discussed before, but I also recognize having 3 filter groups is overwhelming. Perhaps we can leave this feature out for this iteration?

**Comment by arthur791004** [comment]
> Can we functionally treat it the same way the toggle worked but present it as a filter under type?

Currently we treat it the same way the toggle worked. However, users may feel confused about why it works differently from others.

File: packages/design-picker/src/constants.ts (hunk 1 above)
Comment (madhusudhand) [reply packages/design-picker/src/constants.ts]: Thanks @arthur791004 for taking care of this.

**Comment by fditrapani** [comment]
> Currently we treat it the same way the toggle worked. However, users may feel confused about why it works differently from others.

Cool. Thanks for confirming. I'd be surprised if people actually notice the difference :D Where's the best place for me to test the latest work?

**Pull request merged**